src/products/views/ProductList/ProductList.tsx

```diff
@@ -227,29 +234,56 @@ export const ProductList: React.FC<ProductListProps> = ({ params }) => {
       getFilterQueryParam,
       navigate,
       params,
+      keepActiveTab: true,
     });
 
   const handleTabChange = (tab: number) => {
     reset();
-    navigate(
-      productListUrl({
-        activeTab: tab.toString(),
-        ...getFilterTabs()[tab - 1].data,
-      }),
-    );
+    const qs = new URLSearchParams(getFilterTabs()[tab - 1]?.data ?? "");
+    qs.append("activeTab", tab.toString());
+
+    navigate(productListUrl() + qs.toString());
   };
 
   const handleFilterTabDelete = () => {
-    deleteFilterTab(currentTab);
+    deleteFilterTab(tabIndexToDelete);
     reset();
-    navigate(productListUrl());
+
+    // When deleting the current tab, navigate to the All products
+    if (tabIndexToDelete === currentTab) {
+      navigate(productListUrl());
+    } else {
+      // When deleting a tab that is not the current one, only remove the action param from the query
+      delete params.action;
+      // When deleting a tab that is before the current one, decrease the activeTab param by 1
+      params.activeTab =
+        tabIndexToDelete < currentTab ? `${currentTab - 1}` : `${currentTab}`;
```

Comment:
**issue**: this is a but unsafe. Mutating directly object lika query params can cause issues, this is unexpected side-effect. Once you do that, your url isn't sync with the objects you have in the memory

Reply:
you have very similar operations in the other places